core/src/main/java/com/graphhopper/routing/profiles/MappedDecimalEncodedValue.java

```diff
@@ -0,0 +1,53 @@
+package com.graphhopper.routing.profiles;
+
+import com.carrotsearch.hppc.IntIntHashMap;
+
+import java.util.List;
+
+public class MappedDecimalEncodedValue extends IntEncodedValue {
+    private final int toValueMap[];
+    private final IntIntHashMap toStorageMap;
+    private final double precision;
+
+    /**
+     * TODO should we really use precision here or use something like the already used 'factor'?
+     */
+    public MappedDecimalEncodedValue(String name, List<Double> values, double precision, Double defaultValue, boolean store2DirectedValues) {
+        super(name, (int) Long.highestOneBit(values.size()), -1, store2DirectedValues);
+
+        this.precision = precision;
+        // store int-int mapping
+        toValueMap = new int[values.size()];
+        toStorageMap = new IntIntHashMap(values.size());
+
+        int index = 0;
+        for (double val : values) {
```

Comment:
Duplicates checking makes sense. Regarding the sorting: using Collection instead of a List is already possible now.